### PR TITLE
Patch gz-*-vendor packages for Jazzy and Rolling

### PR DESCRIPTION
**Pull request by wentasah**
This is my (incomplete) attempt to compile `rviz2` for Jazzy. It requires patching vendored packages, which is what this PR does. However, it's not sufficient - I get the following error:

```
-- Found gz_math_vendor: 0.0.5 (/nix/store/zncdwq643ism55hpyxlb8rg2zxfcddbz-ros-jazzy-gz-math-vendor-0.0.5-r1/share/gz_math_vendor/cmake)
CMake Error at CMakeLists.txt:65 (find_package):
  By not providing "Findgz-math.cmake" in CMAKE_MODULE_PATH this project has
```

Probably, there is some logic in `ament_vendor`, which we do not patch correctly. It will need more investigation. I'll be mostly off-line next week so if someone wants to continue, they can start here.

### CODE REVIEW

**Comment by bjsowa** [comment]
@wentasah I developed my overrides for vendor packages in Rolling distribution and found out that the nix cmake builder replaces `/opt` for `/var/empty` in `gz_*_vendor-extras.cmake.in` files in configurePhase. This results in wrong paths added to CMAKE_PREFIX_PATH when you invoke `find_package` on vendor packages.

A solution seems to be setting `dontFixCmake` attribute. I pushed my overrides to #429. There still seem to be some references to `/var/empty` in resulting packages that need to be fixed, but I managed to at least build and run rviz2.

**Comment by bjsowa** [comment]
I am able to build all `gz-*-vendor` packages for rolling as well as `rviz2`. However, when I create a dev shell with Gazebo and try running `gz` I get:
```
I cannot find any available 'gz' command:
        * Did you install any Gazebo library?
        * Did you set the GZ_CONFIG_PATH environment variable?
            E.g.: export GZ_CONFIG_PATH=$HOME/local/share/gz
```

It seems that the [dsv hook that modifies GZ_CONFIG_PATH](https://github.com/gazebo-release/gz_sim_vendor/blob/rolling/gz_sim_vendor.dsv.in) is not being called. 

The other problem is that yaml files generated by the package, contain wrong paths with `/opt` replaced by `/var/empty`. For example, this is how the `opt/gz_sim_vendor/share/gz/sim8.yaml` file looks:
```
--- # Subcommands available inside Gazebo Sim
format: 1.0.0
library_name: gz-sim-gz
library_version: 8.3.0
library_path: /nix/store/hcbabap59amid99acq6p9k27z3av3fyl-ros-rolling-gz-sim-vendor-0.1.0-r1/var/empty/gz_sim_vendor/lib/ruby/gz/cmdsim8
commands:
    - sim : Run and manage the Gazebo Simulator.
---
```

**Comment by muellerbernd** [comment]
I have managed to build gz-sim-8 without ROS some time ago (see my overlay [here](https://github.com/muellerbernd/gazebo-sim-overlay)). I had to create some patches that compensated these path issues. Maybe we can take some inspiration from there.

**Comment by wentasah** [comment]
I've updated this to the current develop branch and now it's possible to build `rviz2` for Jazzy and Rolling. It no longer depends on `gz-common-vendor`, which requires insecure freeimage library. I think this can be merged now.

File: distros/jazzy/overrides.nix [hunk 1]
```diff
@@ -31,6 +31,108 @@ in {
...
+  };
+
+  gz-sensors-vendor = lib.patchGzAmentVendorGit rosSuper.gz-sensors-vendor {
+    version = "8.2.0";
+    hash = "sha256-j/8kS+Bvaim2gtsZcp+/u8CAE+N24/5qZhciFR0Q8+M=";
+  };
+
+  gz-sim-vendor = lib.patchGzAmentVendorGit rosSuper.gz-sim-vendor {
+    version = "8.5.0";
+    hash = "sha256-10U8H3/EneLv+zQUGr3mkMPKctDlvtMfMhyQp6lacus=";
```
Comment (hacker1024) [inline distros/jazzy/overrides.nix:118]:
```suggestion
    version = "8.6.0";
    hash = "sha256-zSiPHEh3h2J8hGL342tde5U9FLaGnWs72WD9BqyPf6E=";
```

**Comment by wentasah** [comment]
Rebased to develop and updated the versions. Thanks @hacker1024! Now, jazzy and rolling use the same versions.

**Comment by muellerbernd** [comment]
I have used this PR in a custom shell. 
```nix
# Environment containing basic ROS2 tools
{pkgs ? import ../. {}}:
with pkgs;
with rosPackages.jazzy;
  mkShell {
    nativeBuildInputs = [
      (buildEnv {
        paths = [
          gz-cmake-vendor
          gz-common-vendor
          gz-dartsim-vendor
          gz-fuel-tools-vendor
          gz-gui-vendor
          gz-launch-vendor
          gz-math-vendor
          gz-msgs-vendor
          gz-ogre-next-vendor
          gz-physics-vendor
          gz-plugin-vendor
          gz-rendering-vendor
          gz-sensors-vendor
          gz-sim-vendor
          gz-tools-vendor
          gz-transport-vendor
          gz-utils-vendor
          ros-gz
          ros-core
        ];
      })
    ];
    GZ_CONFIG_PATH = "${gz-tools-vendor}/opt/gz_tools_vendor/share/gz:${gz-gui-vendor}/opt/gz_gui_vendor/share/gz\:${gz-cmake-vendor}/opt/gz_cmake_vendor/share/gz\:${gz-fuel-tools-vendor}/opt/gz_fuel_tools_vendor/share/gz\:${gz-launch-vendor}/opt/gz_launch_vendor/share/gz\:${gz-math-vendor}/opt/gz_math_vendor/share/gz\:${gz-msgs-vendor}/opt/gz_msgs_vendor/share/gz\:${gz-physics-vendor}/opt/gz_physics_vendor/share/gz\:${gz-plugin-vendor}/opt/gz_plugin_vendor/share/gz\:${gz-rendering-vendor}/opt/gz_rendering_vendor/share/gz\:${gz-sensors-vendor}/opt/gz_sensors_vendor/share/gz\:${gz-sim-vendor}/opt/gz_sim_vendor/share/gz\:${gz-transport-vendor}/opt/gz_transport_vendor/share/gz\:${gz-utils-vendor}/opt/gz_utils_vendor/share/gz";
  }
```
Now the `gz` tools can be found, but if I try to start `gz sim` I get the following error:
```
gz sim
<internal:/nix/store/f70cr0j0cr264dl51vw78h01yvay7a0v-ruby-3.1.5/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:136:in `require': cannot load such file -- /nix/store/wcsnnqmgld286prb3yb3710mcqrwkapl-ros-jazzy-gz-sim-vendor-0.0.5-r1/var/empty/gz_sim_vendor/lib/ruby/gz/cmdsim8 (LoadError)
```
there is still the problem with `var/empty` in the yaml files.

**Review by hacker1024 — APPROVED** [review]

**Comment by muellerbernd** [comment]
I think the wrong paths, `/opt` replaced by `/var/empty` in the yaml files are caused by relative paths used in the CMakeLists for the `cmd` of some `gz` libs. Like [here for gz-sim](https://github.com/gazebosim/gz-sim/blob/gz-sim8/src/cmd/CMakeLists.txt?plain=1#L17). Same for:
- gz-transport
- gz-plugin
- gz-gui
I have patches for that in my [gazebo overlay](https://github.com/muellerbernd/gazebo-sim-overlay) and there I can use gazebo harmonic without problem. But here I don't know how to apply patches onto the `gz-*-vendor` packages, because we use `patchGzAmentVendorGit` here. Maybe @wentasah can help here?

**Comment by liarokapisv** [comment]
Thanks for the effort, this works perfectly for me (I had to patch a few irrelevant to this PR packages). Are there any blockers for this PR ?

**Comment by lopsided98** [comment]
This is merged as part of #472

**Pull request closed without merging**